red/nodes/Node.js

```diff
@@ -198,12 +198,22 @@ Node.prototype.error = function(msg) {
 };
 
 Node.prototype.metric = function(msg, eventname, metrics) {
-    metrics = metrics || {};
+    if(metrics && metrics !== null) {
```

Comment:
@hbeeken I had to modify the way metrics are added to the nodes. Beforehand if the metrics were instantiated but they were a primitive, your original code prevented the attachment of all the fields we require. So now we check if we're dealing with an object or primitive.

I also added the recording of <code>metrics.type</code>, see below.

@knolleary Any concerns regarding performance of checking for object? I'm personally not concerned as this is only enabled when metrics are enabled. Thoughts?